src/main/java/com/compilercharisma/chameleonbusinessstudio/service/UserService.java

```diff
@@ -26,6 +29,20 @@ public Mono<UserResponse> getAllUsers() {
         return userRepository.findAllUsers();
     }
 
+    // temporary method by Matt
+    public Mono<Boolean> isRegistered(String email){
+        return getAllUsers()
+            .map(ur -> ur.getUsers())
+            .map(lu -> lu.stream().anyMatch(u -> u.getEmail().equalsIgnoreCase(email)));
+    }
+
+    // temporary method by Matt
+    public Mono<Optional<User>> get(String email){
+        return getAllUsers()
+            .map(ur -> ur.getUsers())
+            .map(lu -> lu.stream().filter(u -> u.getEmail().equalsIgnoreCase(email)).findFirst());
```

Comment:
Here if you wanna get rid of the Optional<> you can put after findFirst() a "orElse(null)"